### ISSUE TITLE
ESLint for collaborators

### ISSUE THREAD

**Issue by psren** [tailwindcss/tailwindcss#93]
Hey there,

i failed in code-style in https://github.com/tailwindcss/tailwindcss/pull/89
Would you mind adding ESlint to the build chain, so collaborators are warned if the're failing?

Maybe we can automate tests and linting and use travisci or smt like that?

**Comment by psren** [comment]
Eslint already has problems with some files if I use a default config.

## Config

```js
module.exports = {
    "env": {
        "browser": false,
        "es6": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            "error",
            2
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "never"
        ]
    }
};
```

## Result

```
/Users/patrickheller/Code/Own/tailwindcss/src/build.js
   8:5   error  Unexpected console statement  no-console
   8:5   error  'console' is not defined      no-undef
  30:11  error  Unexpected console statement  no-console
  30:11  error  'console' is not defined      no-undef
  37:1   error  Unexpected console statement  no-console
  37:1   error  'console' is not defined      no-undef
  44:3   error  Unexpected console statement  no-console
  44:3   error  'console' is not defined      no-undef

/Users/patrickheller/Code/Own/tailwindcss/src/cli.js
   4:8   error  '_' is defined but never used                 no-unused-vars
   7:8   error  'defaultConfig' is defined but never used     no-unused-vars
  17:5   error  Unexpected console statement                  no-console
  17:5   error  'console' is not defined                      no-undef
  18:5   error  'process' is not defined                      no-undef
  21:10  error  'require' is not defined                      no-undef
  27:7   error  'process' is not defined                      no-undef
  36:3   error  'console' is not defined                      no-undef
  36:3   error  Unexpected console statement                  no-console
  44:7   error  Unexpected console statement                  no-console
  44:7   error  'console' is not defined                      no-undef
  46:21  error  Unexpected console statement                  no-console
  46:21  error  'console' is not defined                      no-undef
  49:21  error  'require' is not defined                      no-undef
  49:42  error  '__dirname' is not defined                    no-undef
  59:7   error  'console' is not defined                      no-undef
  59:7   error  Unexpected console statement                  no-console
  60:7   error  'process' is not defined                      no-undef
  63:49  error  '__dirname' is not defined                    no-undef
  65:5   error  'process' is not defined                      no-undef
  76:7   error  Unexpected console statement                  no-console
  76:7   error  'console' is not defined                      no-undef
  77:7   error  'process' is not defined                      no-undef
  82:9   error  'process' is not defined                      no-undef
  87:1   error  Expected indentation of 2 spaces but found 4  indent
  88:1   error  Expected indentation of 0 spaces but found 2  indent
  94:15  error  'process' is not defined                      no-undef
  98:3   error  'process' is not defined                      no-undef

/Users/patrickheller/Code/Own/tailwindcss/src/generators/borderWidths.js
   2:8  error  'defineClass' is defined but never used  no-unused-vars
  47:1  error  'module' is not defined                  no-undef

/Users/patrickheller/Code/Own/tailwindcss/src/generators/floats.js
   7:1  error  Expected indentation of 4 spaces but found 6  indent
   8:1  error  Expected indentation of 4 spaces but found 6  indent
   9:1  error  Expected indentation of 4 spaces but found 6  indent
  10:1  error  Expected indentation of 2 spaces but found 4  indent
  17:4  error  Extra semicolon                               semi

/Users/patrickheller/Code/Own/tailwindcss/src/generators/rounded.js
  26:1  error  'module' is not defined  no-undef

/Users/patrickheller/Code/Own/tailwindcss/src/generators/textStyle.js
  1:8  error  '_' is defined but never used  no-unused-vars

/Users/patrickheller/Code/Own/tailwindcss/src/index.js
   1:8   error  'fs' is defined but never used  no-unused-vars
  27:14  error  'require' is not defined        no-undef
  30:12  error  'require' is not defined        no-undef
  30:26  error  'require' is not defined        no-undef
  31:12  error  'require' is not defined        no-undef
  48:22  error  'require' is not defined        no-undef
  51:1   error  'module' is not defined         no-undef

/Users/patrickheller/Code/Own/tailwindcss/src/lib/evaluateTailwindFunctions.js
  10:44  error  Strings must use singlequote  quotes

/Users/patrickheller/Code/Own/tailwindcss/src/lib/substituteClassApplyAtRules.js
  14:11  error  'options' is assigned a value but never used  no-unused-vars

/Users/patrickheller/Code/Own/tailwindcss/src/lib/substituteFocusableAtRules.js
  1:8   error  '_' is defined but never used                 no-unused-vars
  2:8   error  'postcss' is defined but never used           no-unused-vars
  7:11  error  'options' is assigned a value but never used  no-unused-vars

/Users/patrickheller/Code/Own/tailwindcss/src/lib/substituteHoverableAtRules.js
  1:8   error  '_' is defined but never used                 no-unused-vars
  2:8   error  'postcss' is defined but never used           no-unused-vars
  7:11  error  'options' is assigned a value but never used  no-unused-vars

/Users/patrickheller/Code/Own/tailwindcss/src/lib/substitutePreflightAtRule.js
   6:11  error  'options' is assigned a value but never used   no-unused-vars
  10:1   error  Expected indentation of 8 spaces but found 10  indent
  10:58  error  '__dirname' is not defined                     no-undef
  11:1   error  Expected indentation of 8 spaces but found 10  indent

/Users/patrickheller/Code/Own/tailwindcss/src/lib/substituteScreenAtRules.js
  2:8   error  'postcss' is defined but never used         no-unused-vars
  3:8   error  'cloneNodes' is defined but never used      no-unused-vars
  9:11  error  'rules' is assigned a value but never used  no-unused-vars

/Users/patrickheller/Code/Own/tailwindcss/src/util/defineClasses.js
  1:8  error  'postcss' is defined but never used  no-unused-vars

/Users/patrickheller/Code/Own/tailwindcss/src/util/escapeClassName.js
  2:1   error  Expected indentation of 2 spaces but found 4  indent
  2:43  error  Unnecessary escape character: \-              no-useless-escape
  2:51  error  Strings must use singlequote                  quotes
```

**Comment by davidhemphill** [comment]
I think some light ESLint rules would be nice, but I'd also want to couple that with [Prettier](https://github.com/prettier/prettier) to do most of the heavy-lifting. We'd have to be careful to keep it from affecting things like the `defaultConfig` where we want to keep a consistent style that differs from Prettier's configurable options.

**Comment by davidhemphill** [comment]
Implemented in #95.
